lib/Runtime/Library/ModuleRoot.cpp

```diff
@@ -406,29 +406,29 @@ namespace Js
         return this->GetLibrary()->GetGlobalObject()->GlobalObject::DeleteRootProperty(propertyId, flags);
     }
 
-    BOOL ModuleRoot::HasItem(uint32 index)
+    PropertyQueryFlags ModuleRoot::HasItemQuery(uint32 index)
     {
-        return DynamicObject::HasItem(index)
-            || (this->hostObject && JavascriptOperators::HasItem(this->hostObject, index));
+        return JavascriptConversion::BooleanToPropertyQueryFlags(JavascriptConversion::PropertyQueryFlagsToBoolean((DynamicObject::HasItemQuery(index)))
```

Comment:
Right - inliner will figure that out and optimize if that is still the case.
Not making assumptions about details hidden in the functions makes it more flexible to future changes.

---
In reply to: [106304921](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r106304921) [](ancestors = 106304921)